pkg/apis/serving/v1alpha2/framework_onnx.go

```diff
@@ -36,8 +41,15 @@ func (s *ONNXSpec) GetResourceRequirements() *v1.ResourceRequirements {
 }
 
 func (s *ONNXSpec) GetContainer(modelName string, parallelism int, config *InferenceServicesConfig) *v1.Container {
+	uri, _ := url.Parse(s.StorageURI)
```

Comment:
I'm wondering what the best way to handle `err` is here.

Reply:
I think the validation logic can be in the `Validate` and here we can safely extract the filename
`func (s *ONNXSpec) Validate(config *InferenceServicesConfig) error `.

For onnx the ext should always be `.onnx ` ?

Reply:
it's possible that someone does 
```yaml
storageUri: s3://some_bucket/some/key/path/
```
too though right?